playground/src/syntaxColors.ts

```diff
@@ -0,0 +1,31 @@
+export const syntaxColors: { [name: string]: string } = {
```

Comment:
>syntaxColors [](start = 13, length = 12)

Could these be simple plain old CSS definitions?  It's not clear to me what all this mapping logic is doing exactly.